google/oauth2/service_account.py

```diff
@@ -436,6 +455,26 @@ def refresh(self, request):
                 "domain wide delegation is not supported for non-default universe domain"
             )
 
+        # If a refresh is triggered under the case of credential not expired
+        # that means some abnormal cases proactively trigger this refresh, we
+        # will attemp to fetch trust boundary but tolerant on errors if we
+        # already have cached trust boundaries.
+        if (
+            self.valid
```

Comment:
This attribute is deprecated. New code should use the token_state attribute.

Reply:
Thanks. Got it.